.github/workflows/common.yml

```diff
@@ -72,25 +72,30 @@ jobs:
 
       - name: Run required checks/linters
         env:
+          # use GITHUB_TOKEN because only it has access to GitHub Checks API
           REVIEWDOG_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          # In case the workflow triggered by pull_request event, we want to redirect linter's output to reviewdog.
+          # The selection made through converting available cases to json and then choosing an element depending on logical value (true=1, false=0).
+          GOLANGCI_LINT_REDIRECT: ${{ fromJson('["", "| bin/reviewdog -f=golangci-lint -reporter=github-pr-review -fail-on-error=true"]')[github.event.pull_request != null] }}
+          GO_CONSISTENT_REDIRECT: ${{ fromJson('["", "| bin/reviewdog -f=go-consistent -reporter=github-pr-review -fail-on-error=true"]')[github.event.pull_request != null] }}
```

Comment:
In case the output always redirected to reviewdog, we cannot see the results of linters for the runs caused by non-PR events. First, reviewdog filters results of linters to show only changes related to PR. Second, reviewdog doesn't get id of PR if workflow triggered by non-PR event, so reviedog reports there is no pull request commit.